plugins/bigquery/dbt/adapters/bigquery/connections.py

```diff
@@ -194,6 +195,13 @@ def get_bigquery_credentials(cls, profile_credentials):
         error = ('Invalid `method` in profile: "{}"'.format(method))
         raise FailedToConnectException(error)
 
+    @classmethod
+    @lru_cache()
+    def get_bigquery_defaults(cls):
+        """ Returns (credentials, project_id) """
+        # Cached, because the underlying implementation shells out.
+        return google.auth.default(scopes=cls.SCOPE)
```

Comment:
When I test this locally, `cls.get_bigquery_defaults()` seems to return
```
(<google.oauth2.credentials.Credentials object at 0x112630ca0>, None)
```
resulting in the database remaining `None` below, and returning the following error from the google cloud client:
```
OSError: Project was not passed and could not be determined from the environment.
```

I don't think this is an issue on my end, but curious to hear what might be different, if you've managed to get this working locally.

Reply:
What auth are you using? I'm using ADC — i.e. running `gcloud auth login --update-adc`.

What do you get from:

```python
In [1]: from google.auth import default

In [2]: default()
Out[2]: (<google.oauth2.credentials.Credentials at 0x10a19e850>, '{foo_project}')
```

Reply:
Ah ok, this was definitely on me:
```python
In [1]: from google.auth import default

In [2]: default()
Out[2]: (<google.oauth2.credentials.Credentials at 0x10a19e850>, None)
```
I originally set up local gcloud oauth a _while_ ago, so I updated the auth as you recommended, and the second arg now returns my default project.

Even so, I'm still running into this issue when dbt tries to get information about the `None` database by running `list_None`:
```
google.api_core.exceptions.BadRequest: 400 GET https://bigquery.googleapis.com/bigquery/v2/projects/None/datasets?maxResults=10000: Invalid project ID 'None'. Project IDs must contain 6-63 lowercase letters, digits, or dashes. Some project IDs also include domain name separated by a colon. IDs must start with a letter and may not end with a dash.
```

I think we may need to fill in the missing database/project earlier on—ideally as a `__post_init__` on `BigQueryCredentials`—because dbt needs the database value in more places than just its connection client. This would also be relevant in case users want to make use of the Jinja context variable `target.database`/`target.project`.

Reply:
Great, good point. I pushed a change. Generally I would have that sort of function at the module level; but put them as methods as it seems to be fairly class-focused atm. Let me know which you prefer.

Reply:
What's `list_None`? What's the test you're running locally to test this?

Reply:
Great — thanks for checking @jtcohen6 !